ui.apps/src/main/content/jcr_root/apps/venia/components/commerce/productcarousel/v1/productcarousel/clientlibs/js/product-carousel.js

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+(function () {
+
+	/**
+        *	Product Carousel Component	
+	*/    
+    class ProductCarousel {
+
+		// Initialise the variables
+      constructor() {
+		this._speed             = 300;
+        this._delay             = 0;
+        this._effect            = 'linear';  
+        this._product_carousel_parent = document.querySelector('.product-carousel-parent');
```

Comment:
What happens if you add two product carousel components on the same page? I would rather initialize the class for every component in the page and pass the DOM root node of that component to the constructor. Similar to https://github.com/adobe/aem-core-cif-components/blob/master/ui.apps/src/main/content/jcr_root/apps/venia/components/commerce/product/v1/product/clientlib/js/product.js#L72-L73.

Reply:
👍

Reply:
Root node picked from data attribute  is passed at the time of Initialisation of class.Rest of the elements are picked using root node.